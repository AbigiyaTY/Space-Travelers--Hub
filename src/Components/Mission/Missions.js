import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import fetchMissions from '../../redux/missions/missionsRedux';
import Mission from './Mission';
import './Mission.css';

export default function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionsReducer);
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missions.length]);
  // console.log(missions);
  return (
    <div>
      <table className="mission-table">
        <tbody>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
          {missions.map((mission) => (
            <Mission
              key={mission.mission_id}
              id={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
}
