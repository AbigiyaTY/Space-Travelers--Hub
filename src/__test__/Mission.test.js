import renderer from 'react-test-renderer';
import Missions from './__mock__/Mission';

describe('Missions', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Missions />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
