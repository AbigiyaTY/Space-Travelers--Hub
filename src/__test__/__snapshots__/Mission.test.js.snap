// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
Array [
  <h3>
    Missions
  </h3>,
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      Thaicom
    </div>
  </div>,
]
`;
